### DISCUSSION TITLE
Make it compatible with CloudPanel

### DISCUSSION THREAD

**Discussion by peaklabs-dev** [openappsec/openappsec#236] [Feature Requests]
I know a lot of people using CloudPanel. They are using a custom build of nginx which makes it not work with your amazing software. Can you please make it compatible with the custom nginx version of CloudPanel: https://www.cloudpanel.io/

To check the version of nginx that CloudPanel is using,` nginx -V` can be used.

**Comment by orianelou** [comment]
Hi @ayntk-ai , 
Thank you for your suggestion, we'll look into it!

**Comment by syrabo** [comment]
Thank you for your suggestion.

**Comment by WelshDev** [comment]
I'd also love to see this!

**Comment by devbird007** [comment]
Supporting this feature-request as well. I need it desperately.

**Comment by ngoun** [comment]
This feature would be great! Any considerations about it? +Upvoting!

**Comment by orianelou** [comment]
If you're interested in this feature, please 👍 or upvote this issue to show your support!

**Reply by demorgue** [reply]
hi

I am trying to build the packages myself since CloudPanel uses its own custom nginx

Can we get a list of required dependencies for openappsec that we need to install before building on Debian 12 / Ubuntu 24 ?

**Reply by orianelou** [reply]
Hi @demorgue, 

Before compiling the services, for the open-appsec agent you'll need to ensure the latest development versions of the following libraries and tools:

- Boost
- OpenSSL
- PCRE2
- libxml2
- GTest
- GMock
- cURL
- Redis
- Hiredis
- MaxmindDB
- yq

For compiling the attachment you'll need:

- PCRE
- libxml2
- zlib
- OpenSSL
- Geoip
- Python3
